content/automate/ManageIQ/Transformation/Ansible.class/__methods__/launchplaybookasaservice.rb

```diff
@@ -0,0 +1,48 @@
+module ManageIQ
+  module Automate
+    module Transformation
+      module Ansible
+        class LaunchPlaybookAsAService
+          def initialize(handle = $evm)
+            @handle = handle
+          end
+
+          def target_host(task, transformation_hook)
+            target_host = nil
+            case transformation_hook
+            when 'pre'
+              target_host = task.source
+            when 'post'
+              target_host = @handle.vmdb(:vm).find_by(:id => task.get_option(:destination_vm_id))
+            end
+            target_host
+          end
+
+          def main
+            task = @handle.root['service_template_transformation_plan_task']
+            transformation_hook = @handle.inputs['transformation_hook']
+
+            unless transformation_hook == '_'
+              service_template = task.send("#{transformation_hook}_ansible_playbook_service_template")
+              @handle.log(:info, "Service Template Id: #{service_template.id}")
+              return if service_template.nil?
```

Comment:
if `service_template.nil?` then the previous line `service_template.id` will blow up. A simple fix is `#{service_template.try(:id)}`

Reply:
I removed the `@handle.log` as it is mainly used for debug.